### ISSUE TITLE
Can't display images on MacOS

### ISSUE THREAD

**Issue by thomasd3** [muehlhaus/FSharp.Plotly#31]
When trying to show an image:

>    heatmap |> Chart.ShowAsImage(ImageFormat.PNG)

I get the following exception (MacOS catalina):

```
System.ComponentModel.Win32Exception (13): Permission denied
   at System.Diagnostics.Process.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start(String fileName)
   at FSharp.Plotly.ChartExtensions.Chart.ShowAsImage.Static(ImageFormat format, GenericChart ch)
```

I'm surprised to see a Win32Exception on MacOS as well.

**Comment by kMutagene** [comment]
Hi @thomasd3 , sorry for the late reply.

I guess then all the other display functions such as `Chart.Show` dont work either?

The error seems to be related to your OS as Process.start seems only able to open executables on macOS, while on windows using a file opens it with the default program. Seems like we need to adjust this behaviour on runtime based on the OS. I dont have a mac though, so testing this could prove hard. I will push some changes and release a preview package on nuget, would be cool if you could test it when its done.

**Comment by zyzhu** [comment]
Take a look at this pull in XPlot.  I fixed it a while ago to work with different OS.
https://github.com/fslaborg/XPlot/pull/83/commits/19c9251d76a8c88cfa8cef96a9510ed5b41ed03a#diff-b135992486db43681541285661efff96

**Comment by kMutagene** [comment]
Thanks @zyzhu , this is exacty what i am trying now. However it seems like that xdg-open is not installed on every unix machine. I was testing that with the ubuntu wsl app for windows:

```
root@Surface:/mnt/c/Users/kevin/source/repos/muehlhaus/FSharp.Plotly/bin/FSharp.Plotly/netstandard2.0# cd /tmp
root@Surface:/tmp# URL="./7fd73ea4-eaa0-4ae0-8381-e15d337a4d5b.html"; xdg-open $URL

Command 'xdg-open' not found, but can be installed with:

apt install xdg-utils

root@Surface:/tmp#
```

There seem to be a few command line methods around to open the standard browser on a unix system. Currently i am trying to emulate this command: 

`URL="./7fd73ea4-eaa0-4ae0-8381-e15d337a4d5b.html"; xdg-open $URL || sensible-browser $URL || x-www-browser $URL || gnome-open $URL`

which tries the most common ones with fallback. In my case, the third command works:

```
root@Surface:/tmp# URL="./7fd73ea4-eaa0-4ae0-8381-e15d337a4d5b.html"; xdg-open $URL || sensible-browser $URL || x-www-browser $URL || gnome-open $URL

Command 'xdg-open' not found, but can be installed with:

apt install xdg-utils

Couldn't find a suitable web browser!
Set the BROWSER environment variable to your desired browser.
root@Surface:/tmp#
```

Any thoughts on that?

**Comment by kMutagene** [comment]
I adopted@zyzhu 's code snippet. @thomasd3, the changes are available in v1.2.2 if you want to have a look.
